SouvlakMVP/SouvlakMVP/Graph.cs

```diff
@@ -0,0 +1,553 @@
+﻿namespace SouvlakMVP;
+
+using System.Numerics;
+using System.Linq;
+using indexT = System.Int32;
+using edgeWeightT = System.Single;    // no maidens?
+using edgeCountT = System.Int32;
+
+
+/// <summary>
+/// Class for storing city's map (aka. graph in which vertices = intersections and edges = roads)
+/// </summary>
+public class Graph
+{
+    public class DuplicateEdgeException : Exception
+    {
+        public DuplicateEdgeException() : base("Duplicate edges with different weights are not allowed!") { }
+        public DuplicateEdgeException(string message) : base(message) { }
+    }
+
+
+    public class DuplicateVertexException : Exception
+    {
+        public DuplicateVertexException() : base("Duplicate intersections are not allowed!") { }
+        public DuplicateVertexException(string message) : base(message) { }
+    }
+
+
+    /// <summary>
+    /// Class responsible for holding data about single connection, aka. target's ID and distance 
+    /// </summary>
+    public struct Edge
+    {
+        public indexT targetIdx;
+        public edgeWeightT weight;
+        public edgeCountT count;
+
+        public Edge(indexT targetIdx, edgeWeightT weight, edgeCountT count = 1)
```

Comment:
As for `indexT` (other comment): variable name `targetIdx` -> point that it's a _vertex_ -> `targetVerterIdx`:
```suggestion
        public Edge(indexT targetVertexIdx, edgeWeightT weight, edgeCountT count = 1)
```

Reply:
But there is no point? Cause the only index in the entire solution is an index of a vertex (edges are defined by indices of vertices they connect to)